### PR TITLE
chore(node-package): Use tasks for installing dependencies

### PR DESCRIPTION
**Pull request by icj217**
Spawning a task allows end users to hook into the dependency installation process and increases flexibility

This aligns node and python projects on the same dependency installation API. For node projects, had to explicitly exclude the `install` and `install:ci` tasks from being included in `package.json` scripts to prevent infinite loops.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by icj217** [comment]
Was able to resolve all but 1 of the failing tests. For some reason, when synthesizing a `TypescriptProject`, the `TaskRuntime` component can't find the chalk package. I can't think of any reason for this that makes sense, given that the `TaskRuntime` for the project being created is using the project code and `chalk` is an installed dependency. Hoping someone else w/more familiarity with the codebase might be able to shed some light on this.

```
 FAIL  test/typescript/typescript.test.ts
  ● TypeScriptProject with default settings › compiles

    TypeError: chalk.underline is not a function

      113 |     if (envlogs.length) {
      114 |       this.logDebug(
    > 115 |         chalk.gray(`${chalk.underline("env")}: ${envlogs.join(" ")}`)
          |                             ^
      116 |       );
      117 |     }
      118 |

      at new RunTask (src/task-runtime.ts:115:29)
      at TaskRuntime.runTask (src/task-runtime.ts:76:5)
      at NodePackage.installDependencies (src/javascript/node-package.ts:1450:13)
      at NodePackage.postSynthesize (src/javascript/node-package.ts:863:12)
      at TypeScriptProject.synth (src/project.ts:572:14)
      at Object.<anonymous> (test/typescript/typescript.test.ts:23:13)
```

File: src/javascript/node-package.ts [hunk 1]
```diff
@@ -588,6 +599,20 @@ export class NodePackage extends Component {
...
       this.license = options.license ?? "Apache-2.0";
     }
+
+    this.installTask = project.addTask("install", {
+      description:
+        "Install project dependencies and update lockfile (non-frozen)",
+      exec: this.installAndUpdateLockfileCommand,
+    });
+    // Necessary to prevent infinite loop
+    this.removeScript(this.installTask.name);
```
Comment (mrgrain) [inline src/javascript/node-package.ts:609]: yes that's fine. I'd rather have them filtered out in `renderScripts` if you don't mind making that change

File: src/javascript/node-package.ts [hunk 2]
```diff
@@ -588,6 +599,20 @@ export class NodePackage extends Component {
...
+
+    this.installCiTask = project.addTask("install:ci", {
+      description: "Install project dependencies using frozen lockfile",
+      exec: this.installCommand,
+    });
+    this.removeScript(this.installCiTask.name);
```
Comment (mrgrain) [inline src/javascript/node-package.ts:615]: See above

File: src/task-runtime.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { platform } from "os";
 import { dirname, join, resolve } from "path";
 import * as path from "path";
 import { format } from "util";
-import * as chalk from "chalk";
+import { gray, underline } from "chalk";
```
Comment (mrgrain) [inline src/task-runtime.ts:7]: I came across this elsewhere. I think I need to make a PR forcing this everywhere.

**Review by mrgrain — CHANGES_REQUESTED** [review]
I love this 😍 

Just a small request

File: src/task-runtime.ts (hunk 3 above)
Comment (icj217) [reply src/task-runtime.ts:7]: Glad to hear I'm not crazy :)

**Review by mrgrain — APPROVED** [review]

**Pull request merged**